### PR TITLE
Adding Authentication Siddhi App

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
Adding siddhi app to handle and process authentication data

### Proposed changes in this pull request

- Created siddhi app
    - Create streams to capture wso2 event for authentication data
    - Created aggregations and data tables to process and store them.
- Added widgets to Authentication analytics
    - IsAnalyticsMessages
    - IsAnalyticsHorizontalBarChart

### When should this PR be merged

N/A

### Follow up actions

This is still on progress. Will update other work soon too.

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: features/org.wso2.analytics.is.feature/src/main/siddhi4/README.md [hunk 1]
```diff
@@ -1,5 +1,32 @@
...
 Drop the siddhi files in <SP_HOME>/wso2/worker/deployment/siddhi-files/
-To deploy The To deploy the siddhi4 execution files refer link https://docs.wso2.com/display/SP420/Deploying+Streaming+Applications
\ No newline at end of file
+To deploy The To deploy the siddhi4 execution files refer link https://docs.wso2.com/display/SP420/Deploying+Streaming+Applications
+
+**For authentication Data Analytics:**
+- Download WSO2 stream Processor (This will be referred as <SP_HOME>)
+- Open the file "<SP_Home>/conf/worker/deployment.yaml" and add following datasource:
+    ```
+    - name: IS_DATA_DUMP_DB
```
Comment (Niveathika92) [inline features/org.wso2.analytics.is.feature/src/main/siddhi4/README.md]: How about we change it to IS_ANALYTICS_DB?

File: features/org.wso2.analytics.is.feature/src/main/siddhi4/README.md [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
+    description: The datasource used for dashboard feature
+          jndiConfig:
+            name: jdbc/IS_DATA_DUMP_DB
+            useJndiReference: true
+          definition:
+            type: RDBMS
+            configuration:
+              jdbcUrl: 'jdbc:mysql://localhost/IS_DATA_DUMP_DB?useSSL=false'
```
Comment (Niveathika92) [inline features/org.wso2.analytics.is.feature/src/main/siddhi4/README.md]: This should point to H2 database

File: features/org.wso2.analytics.is.feature/src/main/siddhi4/README.md [hunk 3]
```diff
@@ -1,5 +1,32 @@
...
+              username: root
+              password: IsAnalytics
```
Comment (Niveathika92) [inline features/org.wso2.analytics.is.feature/src/main/siddhi4/README.md]: Let's not put username, password of your local instance. use the H2 database with wso2carbon as username and password

**Review by niveathika — CHANGES_REQUESTED** [review]

**Review by ruwanta — APPROVED** [review]

File: features/org.wso2.analytics.is.feature/src/main/siddhi4/README.md (hunk 1 above)
Comment (ThisaruGuruge) [reply features/org.wso2.analytics.is.feature/src/main/siddhi4/README.md]: Okay. Will change the name.

File: features/org.wso2.analytics.is.feature/src/main/siddhi4/README.md (hunk 2 above)
Comment (ThisaruGuruge) [reply features/org.wso2.analytics.is.feature/src/main/siddhi4/README.md]: Oaky. Will change it.

File: features/org.wso2.analytics.is.feature/src/main/siddhi4/README.md (hunk 3 above)
Comment (ThisaruGuruge) [reply features/org.wso2.analytics.is.feature/src/main/siddhi4/README.md]: Okay.

**Pull request merged**